### PR TITLE
Pas 18 workspaces

### PR DESCRIPTION
**Pull request by pawelkondraciuk**

### CODE REVIEW

File: apps/nestjs-api/src/app/auth-jwt.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+@Injectable()
+export class AuthJwtMiddleware implements NestMiddleware {
+  constructor(
+    private readonly userService: UsersService,
+    private readonly jwtService: JwtService
+  ) {}
+
+  async use(req: Request, res: Response, next: NextFunction) {
+    const authHeaders = req.headers.authorization;
```
Comment (lukszymanski) [inline apps/nestjs-api/src/app/auth-jwt.middleware.ts:16]:
```
const [_, authHeaders] = (req.headers.authorization as string).split(' ');
```

File: apps/nestjs-api/src/app/order/order.controller.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  @Post('/mark-as-ordered')
+  markAsOrdered(
+    @CurrentOrder() order: OrderEntity,
+    @Body() markAsOrderedDto: MarkAsOrderedDto,
+    @CurrentUser() user: UserEntity
+  ) {
+    return this.ordersService.markAsOrdered(order.id, markAsOrderedDto, user);
+  }
+
+  @Post('/mark-as-closed')
```
Comment (lukszymanski) [inline apps/nestjs-api/src/app/order/order.controller.ts:38]: would create a single endpoint with targeted payload intead, what do yoi think?

File: libs/components/src/lib/header-breadcrumbs/header-breadcrumbs-item.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { NavLink, NavLinkProps } from 'react-router-dom';
+import { ChevronRightIcon } from '@heroicons/react/solid';
+
+export type HeaderBreadcrumbsItemProps = NavLinkProps & {
+  children?: JSX.Element | string;
```
Comment (lukszymanski) [inline libs/components/src/lib/header-breadcrumbs/header-breadcrumbs-item.tsx:5]: please use React.FC

File: libs/layout/src/lib/layout.store.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+export const useLayoutStore = create<LayoutStore>((set) => ({
+  addWorkspaceModalOpen: false,
+  sidebarOpen: false,
+
+  showAddWorkspaceModal: () => {
+    console.log('show');
+    set({ addWorkspaceModalOpen: true });
+  },
+  hideAddWorkspaceModal: () => {
+    console.log('close');
```
Comment (lukszymanski) [inline libs/layout/src/lib/layout.store.ts:21]: console.log

File: libs/layout/src/lib/layout.store.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  openSidebar: () => void;
+  closeSidebar: () => void;
+}
+
+export const useLayoutStore = create<LayoutStore>((set) => ({
+  addWorkspaceModalOpen: false,
+  sidebarOpen: false,
+
+  showAddWorkspaceModal: () => {
+    console.log('show');
```
Comment (lukszymanski) [inline libs/layout/src/lib/layout.store.ts:17]: console.log

File: libs/layout/src/lib/layout.tsx [hunk 6]
```diff
@@ -1,27 +1,19 @@
...
-  const closeSidebar = useCallback(() => {
-    setSidebarOpen(false);
-  }, []);
-  const openSidebar = useCallback(() => {
-    setSidebarOpen(true);
-  }, []);
-  return !user ? (
-    <FullscreenSpinner />
-  ) : (
+  const sidebarOpen = useLayoutStore((store) => store.sidebarOpen);
```
Comment (lukszymanski) [inline libs/layout/src/lib/layout.tsx:13]: second parameter of zustand selector is the equality fn could use that to memoize the result

File: libs/pages/dashboard/src/lib/dashboard-order-list/dashboard-order-list.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  orders: OrderModel[];
+}
+
+export function DashboardOrderList({ orders }: DashboardOrderListProps) {
+  return (
+    <StackedList>
+      {orders.map((order) => (
+        <StackedList.Item
+          key={order.id}
+          wrapper={({ children }) => (
```
Comment (lukszymanski) [inline libs/pages/dashboard/src/lib/dashboard-order-list/dashboard-order-list.tsx:24]: useMemo

File: libs/pages/order/src/lib/order-header/order-header.tsx [hunk 8]
```diff
@@ -64,7 +63,24 @@ export function OrderHeader({ order, dishes }: OrderHeaderProps) {
       <div className="px-4 sm:px-6 lg:max-w-6xl md:mx-auto lg:px-8">
         <div className="py-6 xl:flex xl:items-center lg:justify-between lg:border-t lg:border-gray-200">
           <div className="flex-1 min-w-0 ml-3">
-            <h1 className="text-2xl font-semibold leading-7 text-gray-900 sm:leading-9 sm:truncate flex items-center">
+            <HeaderBreadcrumbs
+              back={() => (
```
Comment (lukszymanski) [inline libs/pages/order/src/lib/order-header/order-header.tsx:67]: useCallack here would be nice

File: libs/pages/orders/src/lib/order-list/order-list.tsx [hunk 9]
```diff
@@ -1,61 +1,72 @@
...
-                </div>
-              </div>
-            </div>
-          </NavLink>
-        </li>
+    <StackedList>
+      {orders.map((order) => (
+        <StackedList.Item
+          key={order.id}
+          wrapper={({ children }) => (
```
Comment (lukszymanski) [inline libs/pages/orders/src/lib/order-list/order-list.tsx:24]: useCallback

File: libs/pages/workspace/src/lib/workspace-header/workspace-header.tsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+            )}
+          {workspace && <WorkspaceJoinLeaveButton workspace={workspace} />}
+        </div>
+      </div>
+      {workspace && (
+        <UpdateWorkspaceDrawer
+          workspace={workspace}
+          isOpen={editWorkspace}
+          onSuccess={onUpdateWorkspaceSuccess}
+          onCancel={() => setEditWorkspace(false)}
```
Comment (lukszymanski) [inline libs/pages/workspace/src/lib/workspace-header/workspace-header.tsx:73]: useCalblack

File: libs/pages/workspace/src/lib/workspace-header-users/workspace-header-users.tsx [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+export function WorkspaceHeaderUsers() {
+  const { slug } = useParams<'slug'>();
+  const { data: users } = useWorkspaceUsers(slug!);
+
+  return (
+    <WorkspaceUsers
+      usersToShow={10}
+      avatarSize="sm"
+      users={users}
+      popoverElement={({ workspaceUser }) => (
```
Comment (lukszymanski) [inline libs/pages/workspace/src/lib/workspace-header-users/workspace-header-users.tsx:14]: useCallback mordo

File: libs/pages/workspace/src/lib/workspace-order-list/workspace-order-list.tsx [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+  orders: OrderModel[];
+}
+
+export function WorkspaceOrderList({ orders }: WorkspaceOrderListProps) {
+  return (
+    <StackedList>
+      {orders.map((order) => (
+        <StackedList.Item
+          key={order.id}
+          wrapper={({ children }) => (
```
Comment (lukszymanski) [inline libs/pages/workspace/src/lib/workspace-order-list/workspace-order-list.tsx:23]: useCallback would be nice here

File: libs/pages/workspace/src/lib/workspace-order-list/workspace-order-list.tsx [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+            <NavLink
+              to={`/order/${order.slug}`}
+              className="block hover:bg-gray-50"
+            >
+              {children}
+            </NavLink>
+          )}
+          title={order.from}
+          titleRight={<OrderStatusBadge order={order} />}
+          subTitle={
```
Comment (lukszymanski) [inline libs/pages/workspace/src/lib/workspace-order-list/workspace-order-list.tsx:33]: useMemo in this case

File: libs/pages/workspace/src/lib/workspace-page-inactive-orders/workspace-page-inactive-orders.tsx [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+import { WorkspaceOrderList } from '../workspace-order-list/workspace-order-list';
+import { useEffect } from 'react';
+import { useWorkspaceStore } from '@pasnik/features/workspaces';
+import { CallStateWrapper } from '@pasnik/components';
+import { WorkspaceInactiveOrdersEmpty } from '../workspace-inactive-orders-empty/workspace-inactive-orders-empty';
+
+export function WorkspacePageInactiveOrders() {
+  const { workspace } = useWorkspaceStore();
+  const { fetchInactiveOrders } = useWorkspaceOrdersStore();
+  const [orders, callState] = useWorkspaceOrdersStore(
```
Comment (lukszymanski) [inline libs/pages/workspace/src/lib/workspace-page-inactive-orders/workspace-page-inactive-orders.tsx:11]: could use the equality function to memoize the data

File: libs/layout/src/lib/layout.tsx (hunk 6 above)
Comment (pawelkondraciuk) [reply libs/layout/src/lib/layout.tsx:13]: via docs

It detects changes with strict-equality (old === new) by default, this is efficient for atomic state picks.

File: libs/pages/dashboard/src/lib/dashboard-order-list/dashboard-order-list.tsx (hunk 7 above)
Comment (pawelkondraciuk) [reply libs/pages/dashboard/src/lib/dashboard-order-list/dashboard-order-list.tsx:24]: It is inside loop function, could you provide an example

**Review by lukszymanski — APPROVED** [review]

**Pull request merged**